### PR TITLE
TESTS! INTEGRATION TESTS!

### LINKED ISSUE #27: tests. TESTS

### PR DESCRIPTION
**Pull request by 112batuhan**
closes #27 
The goal is to test if the endpoints work as whole. I didn't go too hard on the assertions and settled on `assert len(response.json()) >= 1` and `assert response.status == 200` for most of the endpoints since FastAPI is doing data validation on pydantic models every response. We use that to validate output shape.

Ready to merge.

### CODE REVIEW

File: app/test/conftest.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+                    "scope": "identify public",
+                },
+        ) as response:
+            return await response.json()
+
+
+async def get_authentication_jwt():
+    access_token = await get_osu_credentials_grant_token()
+    # Can't get /me endpoint to work with client credentials grant type.
+    user_details = await get_user_osu(access_token["access_token"], settings.TEST_USER_ID)
```
Comment (efehana) [inline app/test/conftest.py:35]: Why not mock the response?

File: app/utils/osu_requester.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+            self.test_path, f"{method}-{url.replace("/", "-")}.json")
+        if os.path.exists(file_path):
+            with open(file_path, 'r', encoding='utf-8') as json_file:
+                existing_data = type.model_validate_json(
+                    json_file.read())
+            return existing_data
+        else:
+            authenticator = await LazyTestAuthenticator.get_instance()
+            text = await self.inner_request(method, url, authenticator.as_header(), json)
+            with open(file_path, 'w', encoding='utf-8') as json_file:
```
[the remark this answers is not in the document]
Comment (112batuhan) [reply app/utils/osu_requester.py:69]: I changed the path string to hash so no more injections (It was only used in tests anyway)

File: app/utils/osu_requester.py [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+        if self.test_path is None:
+            text = await self.inner_request(method, url, headers, json)
+            return type.model_validate_json(text)
+
+        os.makedirs(self.test_path, exist_ok=True)
+        file_path = os.path.join(
+            self.test_path, f"{method}-{url.replace("/", "-")}.json")
+        if os.path.exists(file_path):
+            with open(file_path, 'r', encoding='utf-8') as json_file:
```
[the remark this answers is not in the document]
Comment (112batuhan) [reply app/utils/osu_requester.py:62]: I changed the path string to hash so no more injections (It was only used in tests anyway)

**Comment by 112batuhan** [comment]
With new stuff, we can close #29

**Pull request merged**